k8sutils/pkg/workload/ownerreference.go

```diff
@@ -1,36 +1,48 @@
 package workload
 
 import (
-	"errors"
+	"fmt"
 	"strings"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// GetWorkloadFromOwnerReference retrieves both the workload name and workload kind
+// from the provided owner reference.
 func GetWorkloadFromOwnerReference(ownerReference metav1.OwnerReference) (workloadName string, workloadKind WorkloadKind, err error) {
-	ownerName := ownerReference.Name
-	ownerKind := ownerReference.Kind
+
+	workloadName, workloadKind, err = GetWorkloadNameAndKind(ownerReference.Name, ownerReference.Kind)
+	if err != nil {
+		return "", "", err
+	}
+
+	return workloadName, workloadKind, nil
```

Comment:
I think you can do this:
```suggestion
	return GetWorkloadNameAndKind(ownerReference.Name, ownerReference.Kind)
```